scripts/exporter.py

```diff
@@ -323,6 +324,62 @@ def main(self) -> None:
         raise NotImplementedError("Marching cubes not implemented yet.")
 
 
+@dataclass
+class ExportSDFMarchingCubesMesh(Exporter):
+    """Load SDF checkpoint, run marching cubes, extract mesh."""
+
+    resolution: int = 1024
+    """Marching cube resolution."""
+    simplify_mesh: bool = False
+    """Whether to simplify the mesh."""
+    bounding_box_min: Tuple[float, float, float] = (-1.0, -1.0, -1.0)
+    """Minimum of the bounding box."""
+    bounding_box_max: Tuple[float, float, float] = (1.0, 1.0, 1.0)
+    """Maximum of the bounding box."""
+    px_per_uv_triangle: int = 4
+    """Number of pixels per UV triangle."""
+    unwrap_method: Literal["xatlas", "custom"] = "xatlas"
+    """The method to use for unwrapping the mesh."""
+    num_pixels_per_side: int = 2048
+    """If using xatlas for unwrapping, the pixels per side of the texture image."""
+    target_num_faces: Optional[int] = 50000
+    """Target number of faces for the mesh to texture."""
+
+    def main(self) -> None:
+        """Main function."""
+        if not self.output_dir.exists():
+            self.output_dir.mkdir(parents=True)
+
+        _, pipeline, _ = eval_setup(self.load_config)
+
+        CONSOLE.print("Extract mesh with marching cubes and may take a while")
+
+        assert self.resolution % 512 == 0
+        # for sdf we can multi-scale extraction.
+        get_surface_sliding(
+            sdf=lambda x: pipeline.model.field.forward_geonetwork(x)[:, 0].contiguous(),
```

Comment:
This is great for the SDF case, but you could imagine changing the lambda function if we want to do marching cubes on NeRF densities.

Reply:
I think this would require a more significant rewrite to make things work with density. I think it's probably worth doing this in a separate PR as I try to figure out the right way of doing it. I want to do some rewriting of the `base_surface_model` and `sdf_field` components as it's currently a bit unwieldy. Let me know your thoughts. I did change all of the above that you mentioned such as adding in the isosurface_threshold parameter.